src/server/scripts/EasternKingdoms/zone_isle_of_queldanas.cpp

```diff
@@ -88,7 +89,346 @@ class npc_greengill_slave : public CreatureScript
     }
 };
 
+/*######
+## npc_thalorien_dawnseeker
+######*/
+
+enum ThalorienDawnseekerSays
+{
+    SAY_THALORIEN_1 = 0,
+    SAY_THALORIEN_2 = 1,
+    SAY_THALORIEN_3 = 2,
+    SAY_THALORIEN_4 = 3,
+    SAY_THALORIEN_5 = 4,
+    SAY_THALORIEN_6 = 5,
+    SAY_THALORIEN_7 = 6,
+    SAY_THALORIEN_8 = 7,
+    SAY_THALORIEN_9 = 8,
+    SAY_THALORIEN_10 = 9
+};
+
+enum MorlenSays
+{
+    SAY_MORLEN_1 = 0,
+    SAY_MORLEN_2 = 1,
+    SAY_MORLEN_3 = 2,
+    SAY_MORLEN_4 = 3,
+    SAY_MORLEN_5 = 4
+};
+
+enum ThalorienDawnseekerCreatures
+{
+    NPC_THALORIEN_DAWNSEEKER    = 37205,
+    NPC_MORLEN_GOLDGRIP         = 37542,
+    NPC_QUEST_CREDIT            = 37601,
+    NPC_SUNWELL_DEFENDER        = 37211,
+    NPC_SCOURGE_ZOMBIE          = 37538,
+    NPC_GHOUL_INVADER           = 37539,
+    NPC_CRYPT_RAIDER            = 37541
+};
+
+enum ThalorienDawnseekerActions
+{
+    ACTION_START_QUEST = 1
+};
+
+enum ThalorienDawnseekerEvents
+{
+    EVENT_STEP_1    = 1,
+    EVENT_STEP_2    = 2,
+    EVENT_STEP_3    = 3,
+    EVENT_STEP_4    = 4,
+    EVENT_STEP_5    = 5,
+    EVENT_STEP_6    = 6,
+    EVENT_STEP_7    = 7,
+    EVENT_STEP_8    = 8,
+    EVENT_STEP_9    = 9,
+    EVENT_STEP_10   = 10,
+    EVENT_STEP_11   = 11,
+    EVENT_STEP_12   = 12,
+    EVENT_STEP_13   = 13,
+    EVENT_STEP_14   = 14,
+    EVENT_STEP_15   = 15,
+    EVENT_STEP_16   = 16,
+    EVENT_STEP_17   = 17,
+    EVENT_STEP_18   = 18,
+    EVENT_STEP_19   = 19
+};
+
+enum ThalorienDawnseekerMisc
+{
+    OPTION_ID_REMAINS       = 0,
+    DISPLAY_MOUNT           = 25678,
+    QUEST_THALORIEN_A       = 24535,
+    QUEST_THALORIEN_H       = 24563,
+    GOSSIP_MENU_ID_REMAINS  = 37552,
+    SPELL_BLOOD_PRESENCE    = 50689
+};
+
+Position const ZombieLoc[5] =
+{
+    { 11768.7f, -7065.83f, 24.0971f, 0.125877f },
+    { 11769.5f, -7063.83f, 24.1399f, 6.06035f },
+    { 11769.8f, -7061.41f, 24.2536f, 6.06035f },
+    { 11769.4f, -7057.86f, 24.4624f, 0.00335493f },
+    { 11769.4f, -7054.56f, 24.6869f, 0.00335493f }
+};
+
+Position const GuardLoc[4] =
+{
+    { 11807.0f, -7070.37f, 25.372f, 3.218f },
+    { 11805.7f, -7073.64f, 25.5694f, 3.218f },
+    { 11805.6f, -7077.0f, 25.9643f, 3.218f },
+    { 11806.0f, -7079.71f, 26.2067f, 3.218f }
+};
+
+class npc_thalorien_dawnseeker : public CreatureScript
+{
+    public:
+        npc_thalorien_dawnseeker() : CreatureScript("npc_thalorien_dawnseeker") { }
+
+        bool OnGossipHello(Player* player, Creature* creature) override
```

Comment:
OK, I can do that. Just need to finish one of my other PRs and then I'll convert the texts.